x-pack/plugins/watcher/public/sections/watch_edit/components/threshold_watch_edit/threshold_watch_action_dropdown.tsx

```diff
@@ -47,54 +42,75 @@ interface Props {
+      return {
+        type,
+        typeName,
+        iconClass,
+        selectMessage,
+        isEnabled,
+      };
+    }
+  );
+
+  const button = (
+    <EuiButton
+      data-test-subj="createIndexPatternButton"
+      iconType="arrowDown"
+      iconSide="right"
+      onClick={() => setIsPopOverOpen(!isPopoverOpen)}
+    >
+      <FormattedMessage
+        id="xpack.watcher.sections.watchEdit.actions.addActionButtonLabel"
+        defaultMessage="Add action"
+      />
+    </EuiButton>
   );
 
-  const actionOptions = actions
-    ? actions.map((action: any) => {
-        const isActionDisabled = action.type === ACTION_TYPES.EMAIL && !action.isEnabled; // Currently can only fully verify email action
-        const description = isActionDisabled ? disabledMessage : action.selectMessage;
-        return {
-          value: action.type,
-          inputDisplay: action.typeName,
-          disabled: isActionDisabled && !action.isEnabled,
-          dropdownDisplay: (
-            <EuiFlexGroup>
-              <EuiFlexItem grow={false} style={{ alignSelf: 'center' }}>
-                <EuiIcon type={action.iconClass} />
-              </EuiFlexItem>
-              <EuiFlexItem grow={false}>
-                <strong>{action.typeName}</strong>
-                <EuiSpacer size="xs" />
-                <EuiText size="s" color="subdued">
-                  <p className="euiTextColor--subdued">{description}</p>
-                </EuiText>
-              </EuiFlexItem>
-            </EuiFlexGroup>
-          ),
-        };
-      })
-    : [];
-  const actionOptionsWithEmptyValue = [firstActionOption, ...actionOptions];
   return (
-    <EuiSuperSelect
-      options={actionOptionsWithEmptyValue}
-      valueOfSelected={EMPTY_FIRST_OPTION_VALUE}
-      onChange={(value: string) => {
-        addAction({ type: value, defaults: { isNew: true } });
-      }}
-      itemLayoutAlign="top"
-      hasDividers
-      isLoading={isLoading}
-    />
+    <EuiPopover
+      id="singlePanel"
+      button={button}
+      isOpen={isPopoverOpen}
+      closePopover={() => setIsPopOverOpen(false)}
+      panelPaddingSize="none"
+      anchorPosition="downLeft"
+    >
+      <EuiContextMenuPanel
+        items={actions.map((action, index) => {
+          const isActionDisabled = action.type === ACTION_TYPES.EMAIL && !action.isEnabled; // Currently can only fully verify email action
+          const description = isActionDisabled ? disabledMessage : action.selectMessage;
+          return (
+            <EuiContextMenuItem
+              key={`${action.type}-${index}`}
+              disabled={isActionDisabled}
+              onClick={() => {
+                addAction({ type: action.type, defaults: { isNew: true } });
+                setIsPopOverOpen(false);
+              }}
+            >
+              <EuiFlexGroup>
+                <EuiFlexItem grow={false} style={{ alignSelf: 'center' }}>
```

Comment:
We have to avoid inline styles in order to not violate our CSP. Can you create a custom class instead?

Reply:
Good catch - fixed!